scientific/src/commonMain/kotlin/unit/DimensionlessUnit.kt

```diff
@@ -0,0 +1,144 @@
+ * Phrases such as “percentage by mass”, “percentage by volume”, or “percentage by amount of substance” shall not be used;
+ * the extra information on the quantity should instead be conveyed in the description and symbol for the quantity.
+ * "
+ * Source:  "SI Brochure: The International System of Units, 9th Edition".
+ */
+@Serializable
+sealed class Dimensionless : AbstractScientificUnit<PhysicalQuantity.Dimensionless>(), MetricBaseUnit<MeasurementSystem.MetricAndImperial, PhysicalQuantity.Dimensionless> {
+    companion object {
+        operator fun <
+            Unit : ScientificUnit<PhysicalQuantity.Dimensionless>
+            > Number.invoke(unit: Unit) = this.toDecimal()(unit, ::DimensionlessScientificValue)
+
+        operator fun <
+            Unit : ScientificUnit<PhysicalQuantity.Dimensionless>
+            > Decimal.invoke(unit: Unit) = this(unit, ::DimensionlessScientificValue)
+    }
+}
+
+val DimensionlessUnits: Set<Dimensionless> get() = setOf(
+    One,
+    Percent,
+    Permill
+)
+
+@Serializable
+object One : Dimensionless() {
+    const val UNIT_VALUE = 1.0
+    override val symbol: String = ""
+    override val system = MeasurementSystem.MetricAndImperial
+    override val quantity = PhysicalQuantity.Dimensionless
+    override fun fromSIUnit(value: Decimal): Decimal = value
+    override fun toSIUnit(value: Decimal): Decimal = value
+}
+
+val One.constant get() = One.UNIT_VALUE.invoke(One, ::DimensionlessScientificValue)
+
+@Serializable
+object Percent : Dimensionless() {
+    const val PARTS_PER_HUNDRED = 100.0
+    override val symbol: String = "%"
+    override val system = MeasurementSystem.MetricAndImperial
+    override val quantity = PhysicalQuantity.Dimensionless
+    override fun fromSIUnit(value: Decimal): Decimal = value * PARTS_PER_HUNDRED.toDecimal()
+    override fun toSIUnit(value: Decimal): Decimal = value / PARTS_PER_HUNDRED.toDecimal()
+}
+
+@Serializable
+object Permill : Dimensionless() {
+    const val PARTS_PER_THOUSAND = 1000.0
+    override val symbol: String = "‰"
+    override val system = MeasurementSystem.MetricAndImperial
+    override val quantity = PhysicalQuantity.Dimensionless
+    override fun fromSIUnit(value: Decimal): Decimal = value * PARTS_PER_THOUSAND.toDecimal()
+    override fun toSIUnit(value: Decimal): Decimal = value / PARTS_PER_THOUSAND.toDecimal()
+}
+
+/**
+ * Dimensionless scientific value is different from [DefaultScientificValue] when it comes to
+ * decimal representation.
+ * i.e
+ * ```
+ val percent = 12(Percent)
+ print(percent.value) // 12
+ print(percent.decimalValue) // 0.12
+ * ```
+ */
+@Serializable
+data class DimensionlessScientificValue<Unit : ScientificUnit<PhysicalQuantity.Dimensionless>>(
+    override val value: Double,
+    override val unit: Unit
+) : ScientificValue<PhysicalQuantity.Dimensionless, Unit> {
+    constructor(value: Decimal, unit: Unit) : this(value.toDouble(), unit)
+
+    override val decimalValue: Decimal
```

Comment:
Well in that case, we shouldnt use this override. It will cause side effects and make documentation unclear. `decimalValue` is to get the value as a Decimal as opposed to Double. Make an additional property fractionalValue for this purpose please

Reply:
And fractionalDecimalValue to get it as a Decimal

Reply:
How do you see performing `byMultiplying` and `byDividing` on Dimensionless units in that case?
both those operations use `.decimalValue` for operands.